runtime/compiler/control/OptionsPostRestore.cpp

```diff
@@ -331,79 +250,45 @@ J9::OptionsPostRestore::iterateOverExternalOptions()
    {
+#if defined(J9VM_OPT_JITSERVER)
    if (_argIndexUseJITServer >= _argIndexDisableUseJITServer)
       {
-      // Do JITServer init
+      // Needed for GET_OPTION_VALUE_RESTORE_ARGS
+      J9JavaVM *vm = _jitConfig->javaVM;
 
-      if (_argIndexJITServerAddress >= 0)
-         {
-         // set compInfo->getPersistentInfo()->setJITServerAddress
-         }
-
-      if (_argIndexJITServerPort >= 0)
-         {
-         // set compInfo->getPersistentInfo()->setJITServerPort
-         }
-
-      if (_argIndexJITServerTimeout >= 0)
-         {
-         // set compInfo->getPersistentInfo()->setSocketTimeout
-         }
-
-      if ((_argIndexJITServerSSLKey >= 0) && (_argIndexJITServerSSLCert >= 0))
-         {
-         // Init SSL key and cert
-         }
-
-      if (_argIndexJITServerUseAOTCache > _argIndexDisableJITServerUseAOTCache)
-         {
-         // compInfo->getPersistentInfo()->setJITServerUseAOTCache(true);
-         }
-      else
-         {
-         // compInfo->getPersistentInfo()->setJITServerUseAOTCache(false);
-         }
+      // TODO: Determine what other JITServer init is required
 
-      if (_argIndexRequireJITServer > _argIndexDisableRequireJITServer)
+      // Parse common options
+      if (!TR::Options::JITServerParseCommonOptions(vm->checkpointState.restoreArgsList, vm, _compInfo))
          {
-         // compInfo->getPersistentInfo()->setRequireJITServer(true);
-         // compInfo->getPersistentInfo()->setSocketTimeout(60000);
-         }
-
-      if (_argIndexJITServerLogConnections > _argIndexDisableJITServerLogConnections)
-         {
-         TR::Options::setVerboseOption(TR_VerboseJITServerConns);
-         }
-
-      if (_argIndexJITServerAOTmx > 0)
-         {
-         // JITServerAOTCacheMap::setCacheMaxBytes
+         // TODO: Error condition
          }
 
-      if (_argIndexJITServerLocalSyncCompiles > _argIndexDisableJITServerLocalSyncCompiles)
-         {
-         // compInfo->getPersistentInfo()->setLocalSyncCompiles
-         }
+      // Parse local sync compiles
+      TR::Options::JITServerParseLocalSyncCompiles(vm->checkpointState.restoreArgsList,
+                                                   vm,
+                                                   _compInfo,
+                                                   TR::Options::getCmdLineOptions()->getOption(TR_FullSpeedDebug));
 
-      if (_argIndexJITServerAOTCachePersistence > _argIndexDisableJITServerAOTCachePersistence)
+      if (_argIndexJITServerAddress >= 0)
          {
-         // compInfo->getPersistentInfo()->setJITServerUseAOTCachePersistence(true);
-
-         if (_argIndexJITServerAOTCacheDir > 0)
-            {
-            // compInfo->getPersistentInfo()->setJITServerAOTCacheDir(directory);
-            }
+         char *address = NULL;
+         GET_OPTION_VALUE_RESTORE_ARGS(_argIndexJITServerAddress, '=', &address);
+         _compInfo->getPersistentInfo()->setJITServerAddress(address);
```

Comment:
This is arguably the most important option for a JITServer client. Could you please test locally and make sure it works as designed?